storage/range.go

```diff
@@ -430,32 +504,41 @@ func (r *Range) addReadOnlyCmd(method string, args proto.Request, reply proto.Re
 
 	// If read-consistency is set to INCONSISTENT, run directly.
 	if header.ReadConsistency == proto.INCONSISTENT {
+		// But disallow any inconsistent reads within txns.
+		if args.Header().ReadConsistency == proto.INCONSISTENT && args.Header().Txn != nil {
+			reply.Header().SetGoError(util.Errorf("cannot allow inconsistent reads within a transaction"))
+			return reply.Header().GoError()
+		}
 		return r.executeCmd(0, method, args, reply)
+	} else if args.Header().ReadConsistency == proto.CONSENSUS {
+		reply.Header().SetGoError(util.Errorf("consensus reads not implemented"))
+		return reply.Header().GoError()
+	}
+
+	held, prevTerm := r.HasLeaderLease()
+	if !held {
+		// If we don't hold the lease, but nobody else does either, submit
+		// this read-only command as read-write to trigger an election.
```

Comment:
I'm wary that this may have other implications (for example, addReadWriteCmd implies the use of the response cache, which could grow much larger if a large scan happens to trigger an election). We should separate the triggering of elections from addReadWriteCmd.